src/pybind/mgr/rgw/module.py

```diff
@@ -161,7 +191,7 @@ def _cmd_rgw_realm_bootstrap(self,
 
         return HandleCommandResult(retval=0, stdout="Realm(s) created correctly. Please, use 'ceph rgw realm tokens' to get the token.", stderr='')
 
-    def _parse_rgw_specs(self, inbuf: Optional[str] = None) -> List[RGWSpec]:
+    def _parse_rgw_specs(self, inbuf: str) -> List[RGWSpec]:
```

Comment:
In `mgr/orchestrator/module.py` inbuf is always treated as `Optional[str]`. Do you know what's different here?

Reply:
> In `mgr/orchestrator/module.py` inbuf is always treated as `Optional[str]`. Do you know what's different here?

Good observation. In this case, the main inbuf arg is optional:

```
    def _cmd_rgw_realm_bootstrap(self,
                                 realm_name: Optional[str] = None,
                                 zonegroup_name: Optional[str] = None,
                                 zone_name: Optional[str] = None,
                                 port: Optional[int] = None,
                                 placement: Optional[str] = None,
                                 zone_endpoints: Optional[str] = None,
                                 start_radosgw: Optional[bool] = True,
                                 inbuf: Optional[str] = None) -> HandleCommandResult:

```
However, the call to `self._parse_rgw_specs(inbuf)` is only performed if the inbuf is not `None`:

```
        if inbuf:
            try:
                rgw_specs = self._parse_rgw_specs(inbuf)
            except RGWSpecParsingError as e:
                return HandleCommandResult(retval=-errno.EINVAL, stderr=f'{e}')
```

So for the second call we know that the argument is an str and could not be `None`.